src/transformers/integrations/integration_utils.py

```diff
@@ -1005,12 +1017,7 @@ def setup(self, args, state, model):
         self._experiment_name = os.getenv("MLFLOW_EXPERIMENT_NAME", None)
         self._flatten_params = os.getenv("MLFLOW_FLATTEN_PARAMS", "FALSE").upper() in ENV_VARS_TRUE_VALUES
         self._run_id = os.getenv("MLFLOW_RUN_ID", None)
-        self._async_log = False
-        # "synchronous" flag is only available with mlflow version >= 2.8.0
-        # https://github.com/mlflow/mlflow/pull/9705
-        # https://github.com/mlflow/mlflow/releases/tag/v2.8.0
-        if packaging.version.parse(get_mlflow_version()) >= packaging.version.parse("2.8.0"):
-            self._async_log = True
+        self._async_log = is_mlflow_async_log_available()
```

Comment:
```suggestion
        self._async_log = packaging.version.parse(self._mlflow.__version__) >= packaging.version.parse("2.8.0"):
```
should be the only thing you have to do